include/Conversion/QUIRToPulse/QUIRToPulse.h

```diff
@@ -62,9 +62,9 @@ struct QUIRToPulsePass
   mlir::Operation *mainFuncFirstOp;
 
   // convert quir circuit to pulse sequence
-  void convertCircuitToSequence(mlir::quir::CallCircuitOp callCircuitOp,
-                                mlir::func::FuncOp &mainFunc,
-                                ModuleOp moduleOp);
+  mlir::pulse::CallSequenceOp
+  convertCircuitToSequence(mlir::quir::CallCircuitOp callCircuitOp,
+                           mlir::func::FuncOp &mainFunc, ModuleOp moduleOp);
```

Comment:
This is one thing that confuses me about MLIR operations. The `callCircuitOp` and `moduleOp` are not explicitly passed by reference, but the `mainFunc` is. 
Do you know if these are still being passed by reference, or if this is going to pass by value? Pass by value will create copies, which is inefficient. It is better to pass by reference and mark as `const` to indicate they will not be changed.